### ISSUE TITLE
Question: why is only the track_score in the last decoder is kept?

### ISSUE THREAD

**Issue by hilibo** [OpenDriveLab/UniAD#79]
Hi Great work and thanks for opening source the code!

I have a question in `UniADTrack._forward_single_frame_train`. There 6 decoders on different levels when detecting objects from bev features. When constructing track_instances, detections from all 6 decoders are all kept. However, I found that the track scores are all filled as the sigmoid logits of the last decoder, while logits and boxes are filled from each decoder respectively. I wonder if this is an intended inconsistency. Below is the related code.

https://github.com/OpenDriveLab/UniAD/blob/027e31cb0c295bd8cf43455c0218dc02b580e893/projects/mmdet3d_plugin/uniad/detectors/uniad_track.py#L412

https://github.com/OpenDriveLab/UniAD/blob/027e31cb0c295bd8cf43455c0218dc02b580e893/projects/mmdet3d_plugin/uniad/detectors/uniad_track.py#L443-L450

I suppose L447 to be like `track_instances.scores = track_scores[i]`. Please correct me if I misunderstood. Thank you!

**Comment by ilnehc** [comment]
@hilibo Thank you for the question. Track scores are not involved in the loss calculation process for each layer, and only the scores from the last layer are preserved for filtering. As a result, the scores in L447 are just dummy values.

**Comment by ilnehc** [comment]
I will close this issue now. Please reopen it if you need any more help.
